MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationOptionsActivity.java

```diff
@@ -0,0 +1,198 @@
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    LocationOptionsActivity.this.mapboxMap = mapboxMap;
+    enableLocationPlugin();
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  private void enableLocationPlugin() {
+    // Check if permissions are enabled and if not request
+    if (PermissionsManager.areLocationPermissionsGranted(this)) {
+
+      // Set up the location engine
+      initializeLocationEngine();
+
+      // Create and customize the plugin's options
+      LocationLayerOptions locationLayerOptions = LocationLayerOptions.builder(this)
+          .elevation(5)
+          .accuracyAlpha(.6f)
+          .accuracyColor(Color.RED)
+          .foregroundDrawable(R.drawable.android_custom_location_icon)
+          .build();
+
+      // Create the plugin
+      LocationLayerPlugin locationLayerPlugin = new LocationLayerPlugin(
+          mapView, mapboxMap, locationEngine, locationLayerOptions);
+
+      // Set the plugin's camera and render modes
+      locationLayerPlugin.setCameraMode(CameraMode.TRACKING);
+      locationLayerPlugin.setRenderMode(RenderMode.NORMAL);
+
+      // Add the location icon click listener
+      locationLayerPlugin.addOnLocationClickListener(this);
+
+      getLifecycle().addObserver(locationLayerPlugin);
+    } else {
+      permissionsManager = new PermissionsManager(this);
+      permissionsManager.requestLocationPermissions(this);
+    }
+  }
+
+
+  @Override
+  public void onLocationLayerClick() {
+    Toast.makeText(this, "OnLocationLayerClick", Toast.LENGTH_LONG).show();
+  }
+
+  private void initializeLocationEngine() {
+    locationEngine = new LocationEngineProvider(this).obtainBestLocationEngineAvailable();
+    locationEngine.setPriority(LocationEnginePriority.HIGH_ACCURACY);
+    locationEngine.setFastestInterval(1000);
+    locationEngine.addLocationEngineListener(this);
+    locationEngine.activate();
+  }
+
+  @Override
+  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+    permissionsManager.onRequestPermissionsResult(requestCode, permissions, grantResults);
+  }
+
+  @Override
+  public void onExplanationNeeded(List<String> permissionsToExplain) {
+    Toast.makeText(this, R.string.user_location_permission_explanation, Toast.LENGTH_LONG).show();
+  }
+
+  @Override
+  public void onPermissionResult(boolean granted) {
+    if (granted) {
+      enableLocationPlugin();
+    } else {
+      Toast.makeText(this, R.string.user_location_permission_not_granted, Toast.LENGTH_LONG).show();
+      finish();
+    }
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+    if (locationEngine != null) {
+      locationEngine.requestLocationUpdates();
+      locationEngine.addLocationEngineListener(this);
```

Comment:
I don't think we need to add the listener if the engine is already connected, maybe we can check for that?

Reply:
not needed because the locationComponent is handling the engine.